akka-http-core/src/main/scala/akka/http/scaladsl/model/Multipart.scala

```diff
@@ -187,10 +187,7 @@ object Multipart {
 
   private def strictify[BP <: Multipart.BodyPart, BPS <: Multipart.BodyPart.Strict](parts: Source[BP, Any])(f: BP ⇒ Future[BPS])(implicit fm: Materializer): Future[Vector[BPS]] = {
     import fm.executionContext
-    // TODO: move to Vector `:+` when https://issues.scala-lang.org/browse/SI-8930 is fixed
-    parts.runFold(new VectorBuilder[Future[BPS]]) {
-      case (builder, part) ⇒ builder += f(part)
-    }.fast.flatMap(builder ⇒ FastFuture.sequence(builder.result()))
+    parts.mapAsync(Int.MaxValue)(f).runWith(Sink.seq).fast.map(_.toVector)
```

Comment:
Ok, so the deadlock would occur on `Multipart` instances that are created by the user and which have overlapping part streams, I see. Not a very probably scenario but not impossible, true.
The `mapAsync` is definitely the more generally correct solution here, thanks!